src/core/RegisterBase.h

```diff
@@ -208,7 +227,9 @@ const Content & RegisterBase<Content>::get(const std::vector<void*> & images,con
     auto qualified_key=imageToString(*image) + ":" + key;
     if(m.count(qualified_key)>0) return m.find(qualified_key)->second->content;
   }
-  plumed_assert(m.count(key)>0);
+  if (m.count(key) == 0 ) {
+    throw ExceptionRegisterError().setMissingKey(key);
```

Comment:
I set up the additional member because the string ctor (Exception.h:222)
```c++
  explicit Exception(const std::string & msg):
    Exception()
  {
    *this << msg;
  }
```
calls `operator <<` and that prepends a newline to the internal message if `note` is set to true, that is its default value, so `.what()` always starts with `\n`.
I think that manipulating the exception like this is clearer and more flexible like I did in ActionRegister.cpp:49:
```c++
try{
...
} catch (PLMD::ExceptionRegisterError &e ) {
  auto& actionName = e.getMissingKey();
  e <<"Action \"" << actionName << "\" is not known.";
  if (getModuleMap().count(actionName)>0) {
    e << "\nAn Action named \""
      <<actionName
      <<"\" is available in module \""
      << getModuleMap().at(actionName)
      << "\".\nPlease consider installing PLUMED with that module enabled.";
  }
}
```

In this way, I have this error about this "key", and I can react to the key in the catch block, separating the logic of `getModuleMap()` from the logic of the register.

I use `throw ExceptionRegisterError().setMissingKey(key);` three times, so moving the `getModuleMapp()` logic in the catch block prevents me from repeating the code.
This should make the code more flexible because the `throws` are in the RegisteBase and the exception could be thrown from an Action or CL register specialization.

Moreover, in this way, I do not need to remove the eventual stack trace that the ctor puts in `msg` to get the key

If you are ok with this, I think I need to add at least a docstring to explain how to use this for eventual new registers

Reply:
@Iximiel @GiovanniBussi I think this is useful to merge, what is missing?

Reply:
I need to add the docstring and then it is ready